### PR TITLE
Notifications Foundation

### PR DESCRIPTION
**Pull request by andrewmurraydavid**
Implemented System & stubbed CommentViaMention notification types, to show how concrete notifications can be rendered.

Base notification logic (list, mark read, etc) is abstracted from this.

Screenshot 2024-10-21 at 12 23 20 PM
Screenshot 2024-10-21 at 12 22 55 PM
Screenshot 2024-10-21 at 12 22 36 PM

## Release Notes

- **New Features**
  - Introduced a `Notifications` component for managing and displaying user notifications.
  - Added `Notification` and `NotificationViewer` components for rendering individual notifications.
  - Implemented `ReadIndicator` to visually indicate the read status of notifications.
  - Added support for notifications when mentioned in comments through the `CommentViaMention` component.
  - Introduced a `System` component for rendering system notifications with Markdown support.
  - Added new GraphQL queries and mutations for fetching and updating notification statuses, including `NotificationList` and `ReadNotification`.

- **Bug Fixes**
  - Enhanced feature flag handling in the `Feature` component.

- **Documentation**
  - Updated documentation for new components and GraphQL fragments related to notifications.

[Monday Task](https://seed-company-squad.monday.com/boards/5989610236/views/132288519/pulses/7542832529)

### CODE REVIEW

File: src/scenes/Root/Notifications/NotificationList.graphql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+query NotificationList($input: NotificationListInput) {
+  notifications(input: $input) {
+    ...Pagination
+    totalUnread
+    items {
+      ...notification
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (CarsonF) [reply src/scenes/Root/Notifications/NotificationList.graphql:9]: Where are you getting the schema from?

[the remark this answers is not in the document]
Comment (CarsonF) [reply src/scenes/Root/Notifications/NotificationList.graphql:9]: Schema can be introspected from https://api.dev.cordfield.com/graphql
If the PR description has "API PR: {url to PR}" then we pull the unmerged schema file from a build artifact there. I would just ignore these warnings if you see this text in the PR description.

**Comment by andrewmurraydavid** [comment]
@coderabbitai can you please do another full review to see if we missed anything?

File: src/scenes/Root/Notifications/Notifications.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+          notifications: {
+            ...prev!.notifications,
+            totalUnread:
+              prev!.notifications.totalUnread +
+              (updated!.readNotification.unread ? 1 : -1),
+          },
+        })
+      );
+    },
+  });
```
[the remark this answers is not in the document]
Comment (CarsonF) [reply src/scenes/Root/Notifications/Notifications.tsx:51]: Only results returned from the mutation are automatically updated. The mutation does not return the `totalUnread` so it needs to be manually updated.

**Review by CarsonF — APPROVED** [review]
@andrewmurraydavid @rdonigian will you review this?

Notifications can be created with this operation
```gql
mutation {
  createSystemNotification(message: "Hello from Carson") {
    totalRecipients
  }
}
```

**Comment by CarsonF** [comment]
Oh so I did make it optional to have the notification be clickable (a button).
I thought there could be some cases where we don't have a spot to go to or there could be cases where we want to present text with one or more links instead.
I do have the comment via mention as a button.

I don't any automatic mark as read or close notification panel on notification click. Even though both of those things are probably desired in some fashion and cases. 

Lmk y'all's thoughts.

**Comment by andrewmurraydavid** [comment]
LGTM! Works well with the types we have so far! Good work, @CarsonF!

**Comment by CarsonF** [comment]
@rdonigian brought up that we don't have a unread list filter. Let's do that in a separate PR.

**Pull request merged**